extensions/iam/oauth2/oauth2-core/src/main/java/org/eclipse/dataspaceconnector/iam/oauth2/core/impl/Oauth2ServiceImpl.java

```diff
@@ -192,7 +194,15 @@ private JWSVerifier createVerifier(JWSHeader header, String publicKeyId) {
 
     private String buildJwt() {
         try {
-            JWSHeader.Builder headerBuilder = new JWSHeader.Builder(JWSAlgorithm.RS256);
+            JWSHeader.Builder headerBuilder;
+            if (tokenSigner instanceof ECDSASigner) {
```

Comment:
this if could only set the algorithm, that could be a class field, and this assegnation could be done in the constructor, since the signer/algorithm cannot change

Reply:
Done.